core/include/jml/math/activation_functions.hpp

```diff
@@ -12,11 +12,11 @@ class ActivationFunction {
     virtual double f(double x) = 0;
     virtual double df(double x) = 0;
 
-    virtual double f_precomp() = 0;
-    virtual double df_precomp() = 0;
+    virtual double f_precomp();
+    virtual double df_precomp();
```

Comment:
It's just a convenience thing, because the `precomp` functions are only needed for a very few activation functions, so this means that the majority of activation functions which don't use them won't have to declare dummy methods anymore.